files.qmd

```diff
@@ -2,37 +2,74 @@
 
 ## Names
 
-File names should be meaningful and end in `.R`. Avoid using special characters
-in file names - stick with numbers, letters, `-`, and `_`.
-
+1.  File names should be **machine readable**: avoid spaces, symbols, and special characters. Prefer file names that are all lower case, and never
+have names that differ only in their capitalization. Use `.R` as the extension of R files.
 
+    ```
     # Good
     fit_models.R
     utility_functions.R
+    exploratory-data-analysis.R
 
     # Bad
     fit models.R
     foo.r
-    stuff.r
+    ExploratoryDataAnaylsis.r
+    ```
+
+2.  File names should be **human readable**: use file names to describe what's in the file.
+
+    ```
+    # good
+    report-draft-notes.txt
+
+    # bad
+    temp.r
+    ```
+
+    Use the same structure for closely related files:
 
-If files should be run in a particular order, prefix them with numbers. If it
-seems likely you'll have more than 10 files, left pad with zero:
+    ```
+    # good
+    figure-01.png
+    figure-02.png
 
-    00_download.R
-    01_explore.R
-    ...
-    09_model.R
-    10_visualize.R
+    # bad
+    figure 1.png
+    fig 02.png
+    ```
 
-If you later realise that you've missed some steps, it's tempting to use `02a`,
+3.  File names should play well with default ordering: start file names with numbers so that alphabetical sorting puts them in the order they get used.
+
+    ```
+    # good
+    01-load-data.R
+    02-exploratory-analysis.R
+    03-model-approach-1.R
+    04-model-approach-2.R
+
+    # bad
+    alternative model.R
+    code for exploratory analysis.r
+    model_first_try.R
+    run-first.r
+    ```
+
+    If you later realise that you've missed some steps, it's tempting to use `02a`,
 `02b`, etc. However, I think it's generally better to bite the bullet and
 rename all files.
```

Comment:
is indentation messed up here? not sure if it affects the final result

Reply:
It looks ok to me